src/utils/AlertsUtils.js

```diff
@@ -21,29 +21,31 @@ async function fetchAlerts() {
                 },
         };
 
-        const alertsRequest = JSON.parse(await new Promise((resolve, reject) => {
+        const alertsRequest = await new Promise((resolve, reject) => {
             request(options, (error, response, body) => {
                 if (error) reject(error);
                 resolve(body);
             });
         }).then(value => value).catch((error) => {
-            ErrorUtils.log(error, null, 'Alerts request failed');
+            ErrorUtils.log(error, null, 'alerts request failed');
             return null;
-        }));
+        });
 
-        alerts = alertsRequest.map(alert => ({
-            id: alert.MessageId,
-            message: alert.Message,
-            fromDate: parseMicrosoftFormatJSONDate(alert.FromDate),
-            toDate: parseMicrosoftFormatJSONDate(alert.ToDate),
-            fromTime: parseMicrosoftFormatJSONDate(alert.FromTime),
-            toTime: parseMicrosoftFormatJSONDate(alert.ToTime),
-            priority: alert.Priority,
-            daysOfWeek: getWeekdayString(alert.DaysOfWeek),
-            routes: alert.Routes,
-            signs: alert.Signs,
-            channelMessages: alert.ChannelMessages,
-        }));
+        if (alertsRequest) {
+            alerts = JSON.parse(alertsRequest).map(alert => ({
+                id: alert.MessageId,
+                message: alert.Message,
+                fromDate: parseMicrosoftFormatJSONDate(alert.FromDate),
+                toDate: parseMicrosoftFormatJSONDate(alert.ToDate),
+                fromTime: parseMicrosoftFormatJSONDate(alert.FromTime),
+                toTime: parseMicrosoftFormatJSONDate(alert.ToTime),
+                priority: alert.Priority,
+                daysOfWeek: getWeekdayString(alert.DaysOfWeek),
+                routes: alert.Routes,
+                signs: alert.Signs,
+                channelMessages: alert.ChannelMessages,
```

Comment:
similar comment here, can this be alphabetized?

Reply:
I think alphabetizing keys adds a lot of needless complexity and is difficult to maintain. I can make an ESlint rule to alphabetize everything automatically if you want, but I also don't think alphabetizing is better from a readability perspective. 

In this case for example, the first key is "id" which makes sense because that's the alert identifier and the second is "message" because that's the alert text. Then there are four time/date keys to specify the alert duration. Lastly there are optional meta-information keys that the client may not use like alert priority, routes, and signs. I'd personally rather have the data ordered like this instead of alphabetically which is completely arbitrary in terms of grouping and importance.

https://eslint.org/docs/rules/sort-keys
https://eslint.org/docs/rules/sort-vars